entities/StandardGameState.cs

```diff
@@ -0,0 +1,313 @@
+        /// </returns>
+        private static StandardBoard PerformCastle(StandardBoard board, 
+                                                   Move move)
+        {
+
+            // Get a new board.
+            var nextBoard = board.DeepCopy();
+
+            var deltaX = move.EndCell.X - move.StartCell.X;
+            var y = board[move.StartCell]?.Color == Color.White ? 0 : 7;
+            int nextXKing;
+            int nextXRook;
+            int xRook;
+
+            // Short/king castle.
+            if (deltaX > 0)
+            {
+                nextXKing = 6;
+                nextXRook = 5;
+                xRook = 7;
+            }
+            // Long/queen castle.
+            else
+            {
+                nextXKing = 2;
+                nextXRook = 3;
+                xRook = 0;
+            }
+
+            var nextKingCell = new Cell(nextXKing, y);
+            var nextRookCell = new Cell(nextXRook, 
+                                        y);        
+            var rookCell = new Cell(xRook, y);
+            // Perform castle.
+            var kingPiece = nextBoard[move.StartCell];
+            var rookPiece = nextBoard[rookCell];
+            nextBoard[move.StartCell] = null;
+            nextBoard[rookCell] = null;
+            nextBoard[nextKingCell] = kingPiece;
+            nextBoard[nextRookCell] = rookPiece;
+
+            return nextBoard;
+        }
+
+        /// <summary>
+        /// Perform en passant move. The en passant move must be possible/valid.
+        /// </summary>
+        /// <param name="board">
+        /// The start/initial board.
+        /// </param>
+        /// <param name="move">
+        /// The move to perform.
+        /// </param>        
+        /// <returns>
+        /// The next board with performed move. 
+        /// </returns>
+        private static StandardBoard PerformEnPassantMove(StandardBoard board, 
+                                                          Move move)
+        {
+            // Get a new board.
+            var nextBoard = board.DeepCopy();
+            var piece = (Piece)board[move.StartCell];
+
+            // Move own pawn.
+            nextBoard[move.EndCell] = piece;   
+            nextBoard[move.StartCell] = null; 
+
+            // Capture enemy pawn.
+            var enemyCellWithPawn = new Cell(move.EndCell.X, move.StartCell.Y);
+            nextBoard[enemyCellWithPawn] = null;
+
+            return nextBoard;
+        }
+
+        /// <summary>
+        /// Perform the simple move. The move must be possible/valid.
+        /// </summary>
+        /// <param name="board">
+        /// The start/initial board.
+        /// </param>
+        /// <param name="move">
+        /// The move to perform.
+        /// </param>        
+        /// <returns>
+        /// The next board with performed move. 
+        /// </returns>
+        private static StandardBoard PerformMove(StandardBoard board, 
```

Comment:
maybe rename to MakeMove

Reply:
I prefer PerformMove. @amirov-m please choose either PerformMove or MakeMove.